src/main.cpp

```diff
@@ -5607,6 +5607,7 @@ bool static ProcessMessage(const CChainParams& chainparams, CNode* pfrom, string
         if (pfrom->nVersion != 0)
         {
             pfrom->PushMessage("reject", strCommand, REJECT_DUPLICATE, string("Duplicate version message"));
+            LOCK(cs_main);
             Misbehaving(pfrom->GetId(), 1);
```

Comment:
Not for this PR, but I think it would be better for `Misbehaving` to take the lock. The comment on that procedure says that it needs the lock, I can't see why the lock should be taken in this ad-hoc fashion.